connman/plugins/sailfish_ofono.c

```diff
@@ -695,6 +801,12 @@ static void connctx_update_active(struct modem_data *md)
 			if (md->network) {
 				connman_network_set_connected(md->network,
 								FALSE);
+				if (md->delayed_connect_id) {
```

Comment:
I would do the cleanup before calling `connman_network_set_connected()`, and generally try be in a consistent state when calling any connman core function. Just in case if connman feels like calling me back before returning from the function.

Reply:
Done, added a new function for that to be a lot cleaner.